lib/ansible/playbook/base.py

```diff
@@ -440,7 +440,17 @@ def get_validated_value(self, name, attribute, value, templar):
         if attribute.isa == 'string':
             value = to_text(value)
         elif attribute.isa == 'int':
-            value = int(value)
+            msg = ''
+            if isinstance(value, integer_types):
+                pass  # nothing to do here
+            elif isinstance(value, float):
+                msg = "a 'float', which we cannot cast as it would truncate"
+            elif isinstance(value, string_types) and value.lstrip('-').isdigit():
+                value = int(value)
+            else:
+                msg = f"'{value}' of type {type(value)} which is not cleanly castable into an int"
+            if msg:
+                raise AnsibleParserError(f"The keyword '{name}' is required to be of type 'int', but got {msg}", obj=self.get_ds())
```

Comment:
Thinking about it a little more, we probably only want to fail on floating-point values that actually have a mantissa. Otherwise, this change will needlessly break anyone feeding one of these keys from, eg, a naive templated division operation (since the result of Python `/` is always `float`, regardless of input). 

There are a number of scenarios where the proposed impl will raise `ValueError`- something like this should cover all the zero-mantissa and very large value cases nicely, while still failing gracefully when a truncation would occur:

```suggestion
            try:
                if not isinstance(value, int):
                    if (decimal_value := decimal.Decimal(value)) != (int_value := int(decimal_value)):
                        raise ValueError(f'Floating-point value {value!r} would be truncated.')
        
                    value = int_value
            except Exception as ex:
                raise AnsibleParserError(f"Keyword {name!r} failed to convert {value!r} to an integer. {ex}", obj=self.get_ds()) from ex

            return value
```